### PR TITLE
Use fixture submodule for unit tests

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Parse fixture JSON from submodule into dictionaries
- Unit tests updated to use new fixture data
- Tweaked some bad unit test values
- Re-recorded all cassettes

# Testing

- All unit tests updated to use new fixture solution
- Fixed some bad unit test data
- Re-recorded all cassettes since many fixtures changed.

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Comment by Justintime50** [comment]
It doesn't make sense for us to do this in parallel with the rewrite as the work will just need to get redone later.

File: EasyPost.Tests/CarrierAccountTest.cs
Comment (jchen293) [inline EasyPost.Tests/CarrierAccountTest.cs:32]: nit: delete extra line

File: EasyPost.Tests/Fixtures.cs
Comment (jchen293) [inline EasyPost.Tests/Fixtures.cs:51]: nit: add a comment here `If you need to re-record cassettes, increment the date below and ensure it is one day in the future,`

File: EasyPost.Tests/BillingTest.cs
Comment (nwithan8) [reply EasyPost.Tests/BillingTest.cs:12]: Resolved.

File: EasyPost.Tests/CarrierAccountTest.cs
Comment (nwithan8) [reply EasyPost.Tests/CarrierAccountTest.cs:32]: Resolved.

File: EasyPost.Tests/Fixtures.cs
Comment (nwithan8) [reply EasyPost.Tests/Fixtures.cs:51]: Resolved.

File: EasyPost.Tests/AddressTest.cs
Comment (nwithan8) [reply EasyPost.Tests/AddressTest.cs:38]: Resolved.

**Review by jchen293 — DISMISSED** [review]

**Review by Justintime50 — COMMENTED** [review]
I feel like I've looked over these already so much with the rewrite 😂 Can we merge that and then just use the fixtures on the v4 branch instead once landed? I see no benefit do double dipping if we'll have to re-review for v4 here shortly and it'll get replaced with that eventually.

**Comment by nwithan8** [comment]
> I feel like I've looked over these already so much with the rewrite 😂 Can we merge that and then just use the fixtures on the v4 branch instead once landed? I see no benefit do double dipping if we'll have to re-review for v4 here shortly and it'll get replaced with that eventually.

The v4 and rewrite branches are not using the submodule. That's what this is for.

**Comment by Justintime50** [comment]
> The v4 and rewrite branches are not using the submodule. That's what this is for.

Right, I'm suggesting we save us all some work by only doing it on the v4 branch so that when that gets merged to master, we only needed to do the work once instead of twice. At present, the master branch is going to get this change but we'll need to make the same changes on the v4 branch to catch it up.

**Comment by nwithan8** [comment]
> > The v4 and rewrite branches are not using the submodule. That's what this is for.
> 
> Right, I'm suggesting we save us all some work by only doing it on the v4 branch so that when that gets merged to master, we only needed to do the work once instead of twice. At present, the master branch is going to get this change but we'll need to make the same changes on the v4 branch to catch it up.

Understood, we'll hold off on this until the rewrite thread-safe PR is adopted.

File: Makefile
Comment (Justintime50) [inline Makefile:48]: Nit: only one line break

File: EasyPost.Tests/FixtureData/Components/PageSizes.cs
Comment (Justintime50) [inline EasyPost.Tests/FixtureData/Components/PageSizes.cs:17]: Nit: Need line break here

File: EasyPost.Tests/FixtureData/Components/Addresses.cs
Comment (Justintime50) [inline EasyPost.Tests/FixtureData/Components/Addresses.cs:1]: Thought: Do we need these all in separate files/does it make sense? If not, let's consolidate since it seems we aren't really using a separate namespace for fixture data and can reduce a bunch of the complexity by having all fixture definitions in one file like the other libs have. We don't at the end of the day have too many of them.

Comment (nwithan8) [reply EasyPost.Tests/FixtureData/Components/Addresses.cs:1]: We can certainly put them all in one file. I was just replicating Java, which by convention requires them to all be in separate files.

File: Makefile
Comment (nwithan8) [reply Makefile:48]: Resolved

File: EasyPost.Tests/FixtureData/Components/Addresses.cs
Comment (Justintime50) [reply EasyPost.Tests/FixtureData/Components/Addresses.cs:1]: Which of the two options feels most appropriate? I'm good for either.

File: EasyPost.Tests/FixtureData/Components/PageSizes.cs
Comment (nwithan8) [reply EasyPost.Tests/FixtureData/Components/PageSizes.cs:17]: Resolved

File: EasyPost.Tests/FixtureData/Components/Addresses.cs
Comment (nwithan8) [reply EasyPost.Tests/FixtureData/Components/Addresses.cs:1]: Combined, resolved.

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**
